docs/source/managed-federation/deployment.mdx

```diff
@@ -117,11 +118,95 @@ The next time it starts up or polls, your router obtains an updated configuratio
 
 ## Using variants to control rollout
 
-With managed federation, you can control which version of your graph a fleet of routers are using. In the majority of cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
+Your deployment model may require deploying a specific version of your graph, for example for blue-green deployments or canary deployments. With managed federation, you can control which version of your graph a fleet of routers are using. In the majority of cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
 
 However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage different fleets of routers running with different configurations.
 
-### Example
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning the supergraph schema of each environment to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and do analysis on the combined production traffic of all environments.
+
+### Example blue-green deployment
+
+A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic, while the other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.  
+
+As an example, follow these steps to deploy with a supergraph schema of a new release (green) environment:
+
+1. Publish all subgraphs of a release at once by using the Platform API [`publishSubgraphs`](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation#publishSubgraphs).
+
+    ```graphql
+    ## Publish multiple subgraphs together in a batch
+    ## and retrieve the associated launch, along with any downstream launches.
+    mutation PublishSubgraphsMutation(
+      $graphId: ID!
+      $graphVariant: String!
+      $revision: String!
+      $subgraphInputs: [PublishSubgraphsSubgraphInput!]!
+    ) {
+      graph(id: $graphId) {
+        publishSubgraphs( #highlight-line
+          graphVariant: $graphVariant
```

Comment:
Wanted to call out that the variant input argument is called `name` in the other queries below. Not sure if for consistency we want to use one or the other.

Reply:
This would need to be changed in the [publishSubgraphs](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation?_gl=1%2A1tgb7vy%2A_ga%2AOTA0MTI5NDAyLjE2OTk2NTcyMzg.%2A_ga_0BGG5V2W2K%2AMTY5OTkwNTY3Mi4yLjAuMTY5OTkwNTY3Mi4wLjAuMA..#publishSubgraphs) API itself?

Reply:
Not necessarily the API can stay the same but we can change the variable names so that they can be shared between all three operations.

For example:

Variables

```json
{
  "graphId" :"my-graph",
  "graphVariant": "current"
}
```

And the operations would all use `$graphVariant` instead of `$name`

PublishSubgraphsMutation would stay as you have it

```graphql
 query PollLaunchStatusQuery($graphId: ID!, $graphVariant: String!, $launchId: ID!) {
      graph(id: $graphId) {
        variant(name: $graphVariant) {
 ...
```

```graphql
query FetchSupergraphSDLQuery($graphId: ID!, $graphVariant: String!, $launchId: String!) {
      graph(id: $graphId) {
        variant(name: $graphVariant) {
...
```

Reply:
I noticed in the current state of the docs `FetchSupergraphSDLQuery` is missing the `$name`/`$graphVariant` argument in the `query`

Reply:
IMO we should use `graphVariant` instead of `name` 👍